sktime/classification/interval_based/_risf.py

```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
+"""Random Interval Spectral Ensemble (RISE)."""
+
+__author__ = ["Yi-Xuan Xu"]
+
+__all__ = ["RandomIntervalSpectralForest"]
+
+import numpy as np
+from deprecated.sphinx import deprecated
+from joblib import Parallel, delayed
+from sklearn.ensemble._base import _partition_estimators
+from sklearn.ensemble._forest import ForestClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.utils.multiclass import class_distribution
+from sklearn.utils.validation import check_random_state
+
+from sktime.classification.base import BaseClassifier
+from sktime.classification.interval_based._rise import (
+    _make_estimator,
+    _parallel_build_trees,
+    _predict_proba_for_estimator,
+    _select_interval,
+)
+from sktime.utils.validation.panel import check_X, check_X_y
+
+
+@deprecated(
+    version="0.8.1",
+    reason="RandomIntervalSpectralForest will be moved or removed in v0.10.0, "
+    "to be replaced by the correctly named RandomIntervalSpectralEnsemble",
+    category=FutureWarning,
+)
+class RandomIntervalSpectralForest(ForestClassifier, BaseClassifier):
+    """Random Interval Spectral Forest (RISF).
+
+    Input: n series length m
+    for each tree
+        sample a random intervals
+        take the ACF and PS over this interval, and concatenate features
+        build tree on new features
+    ensemble the trees through averaging probabilities.
+
+    Parameters
+    ----------
+    n_estimators : int, optional (default=200)
+        The number of trees in the forest.
+    min_interval : int, optional (default=16)
+        The minimum width of an interval.
+    acf_lag : int, optional (default=100)
+        The maximum number of autocorrelation terms to use.
+    acf_min_values : int, optional (default=4)
+        Never use fewer than this number of terms to find a correlation.
+    n_jobs : int or None, optional (default=None)
+        The number of jobs to run in parallel for both `fit` and `predict`.
+        ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors.
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    Attributes
+    ----------
+    n_classes : int
+        The number of classes, extracted from the data.
+    n_estimators : array of shape = [n_estimators] of DecisionTree classifiers
+    intervals : array of shape = [n_estimators][2]
+        Stores indexes of start and end points for all classifiers.
+    """
+
+    # Capability tags
+    capabilities = {
```

Comment:
should this maybe be repaced with the `capabilities` tags (actual tags), given that this file is new?

Reply:
I'm leaving risf for another day, it does not comply to the interface and is not mine, but will sort it. its a new file but access is the same through init.